File: auth-service/src/main/java/com/ctuconnect/service/AuthService.java
package com.ctuconnect.service;

import com.ctuconnect.dto.AuthResponse;
import com.ctuconnect.dto.LoginRequest;
import com.ctuconnect.dto.RegisterRequest;

public interface AuthService {

    /**
     * Register a new user
     */
    AuthResponse register(RegisterRequest request);

    /**
     * Authenticate a user and return tokens
     */
    AuthResponse login(LoginRequest request);

    /**
     * Refresh an access token using a refresh token
     */
    AuthResponse refreshToken(String refreshToken);

    /**
     * Send a password reset email
     */
    void forgotPassword(String email);

    /**
     * Reset a password using a token
     */
    void resetPassword(String token, String newPassword);

    /**
     * Verify a user's email
     */
    void verifyEmail(String token);

    /**
     * Logout a user by invalidating their refresh token
     */
    void logout(String refreshToken);
}

File: client-frontend/src/features/auth/services/auth.service.ts
import { apiClient } from '@/shared/config/api-client';
import { API_ENDPOINTS } from '@/shared/constants';
import {
  LoginRequest,
  RegisterRequest,
  AuthResponse,
  ForgotPasswordRequest,
  ResetPasswordRequest,
  VerifyEmailRequest,
  ApiResponse,
  User,
} from '@/shared/types/auth';

/**
 * Authentication Service
 * Handles all authentication related API calls
 */
export class AuthService {
  /**
   * Login user
   */
  async login(credentials: LoginRequest): Promise<AuthResponse> {
    return apiClient.post<AuthResponse>(API_ENDPOINTS.AUTH.LOGIN, credentials);
  }

  /**
   * Register new user
   */
  async register(userData: RegisterRequest): Promise<AuthResponse> {
    return apiClient.post<AuthResponse>(API_ENDPOINTS.AUTH.REGISTER, userData);
  }

  /**
   * Logout user
   */
  async logout(): Promise<void> {
    return apiClient.post(API_ENDPOINTS.AUTH.LOGOUT);
  }

  /**
   * Refresh authentication token
   */
  async refreshToken(): Promise<AuthResponse> {
    return apiClient.post<AuthResponse>(API_ENDPOINTS.AUTH.REFRESH);
  }

  /**
   * Get current user profile
   */
  async getCurrentUser(): Promise<ApiResponse<User>> {
    return apiClient.get<User>(API_ENDPOINTS.AUTH.ME);
  }

  /**
   * Verify email with token
   */
  async verifyEmail(request: VerifyEmailRequest): Promise<ApiResponse<null>> {
    return apiClient.post<ApiResponse<null>>(API_ENDPOINTS.AUTH.VERIFY_EMAIL, request);
  }

  /**
   * Send forgot password email
   */
  async forgotPassword(request: ForgotPasswordRequest): Promise<ApiResponse<null>> {
    return apiClient.post<ApiResponse<null>>(API_ENDPOINTS.AUTH.FORGOT_PASSWORD, request);
  }

  /**
   * Reset password with token
   */
  async resetPassword(request: ResetPasswordRequest): Promise<ApiResponse<null>> {
    return apiClient.post<ApiResponse<null>>(API_ENDPOINTS.AUTH.RESET_PASSWORD, request);
  }
}

// Export singleton instance
export const authService = new AuthService();
